File: src/main/java/com/ny/cfss/notes/edm/programs/Enrollment.java
package com.ny.cfss.notes.edm.programs;

import com.ny.cfss.notes.edm.AbstractDomainEntity;
import com.ny.cfss.notes.edm.individual.Individual;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.time.LocalDate;

@Entity
@Table(name = "enrollments")
@Data
@EqualsAndHashCode(callSuper = true, onlyExplicitlyIncluded = true)
public class Enrollment extends AbstractDomainEntity {

    @ManyToOne
    @JoinColumn(name = "programId")
    private Program program;

    @ManyToOne
    @JoinColumn(name = "individualId")
    private Individual individual;

    private LocalDate startDate;
    private LocalDate endDate;

}

File: client/src/app/individuals/enrollment.ts
import {DomainEntity} from "../shared/domain/domain-entity";
import {Individual} from "../individuals/individual";

export class Enrollment extends DomainEntity {

    individual: Individual;
    startDate: Date;
    endDate: Date;

    resourceName(): string {
        return "enrollments";
    }

    isActive(): boolean {
        return this.endDate == null;
    }

}
